#!/usr/bin/env python

PACKAGE = 'skid_steering'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

integration_enum = gen.enum([ gen.const("Euler",      int_t, 0, "Euler Integration"),
                       gen.const("Runge_Kutta",     int_t, 1, "Runge-Kutta Integration")],
                     "An enum to select integration method")

gen.add("enable_odom_tf", bool_t, 0, "Publish odom frame to tf.", True)
gen.add("integration_method", int_t, 0, "An integration parameter which is edited via an enum", 0, 0, 1, edit_method=integration_enum)


exit(gen.generate(PACKAGE, "skid_steering", "SkidSteering"))